content/reference/wadm/api.md

```diff
@@ -0,0 +1,252 @@
+
+The response indicates success or failure, and if an `undeploy` operation was requested as part of the delete.
+
+**Request**: 
+```json
+{
+    "version": "1.0",
+    "delete_all": false,
+    "undeploy": false
+}
+```
+
+**Response**:
+```json
+{
+    "result": "deleted|error",
+    "undeploy": true,
+    "message": "..."
+}
+```
+
+### Version History
+`wadm.api.{lattice}.model.versions.{name}`
+
+Retrieves a list of all stored versions of a given model, including when that version was created. 
+
+**Request**: empty
+
+**Response**: 
+```json
+[
+    {
+        "version": "...",
+        "created": "<RFC3339 Timestamp>",
+        "deployed": false
+    }
+]
+```
+
+Each of the items in the response list describes a revision and indicates whether that revision is the deployed one.
+
+## Managing Deployments
+Deployments are discrete instances of autonomous control agents that monitor an application model against the real-time, discovered state of a lattice. This agent monitors lattice events, compares lattice state against the desired state of the model, and issues lattice control interface commands in order to compensate for observed gaps.
+
+### Deploy
+`wadm.api.{lattice}.model.deploy.{name}`
+
+Deploys the indicated model. This operation is idempotent, and will simply "succeed" if a request to deploy an already deployed model is received. The previous version, if deployed, does not need to be _undeployed_. Instead, the desired state of the new version is compared against the observed state of the lattice and the autonomous agents will compensate accordingly.
+
+**Request**:
+```json
+{
+    "version": "..version to deploy"
+}
+```
+
+**Response**:
+```json
+{
+    "result": "success|error",
+    "message": "..."
+}
+```
+
+It's important to note that the success/fail of this call doesn't indicate completed deployment, only success or failure of the receipt of the deployment request. In other words, the response is an _ack_.
+
+### Undeploy
+`wadm.api.{lattice}.model.undeploy.{name}`
+
+Undeploys the currently deployed version of a model. If the model is already undeployed, the operation is idempotent and will return success along with an appropriate message. If the request field `destructive` is true, then the agent will attempt to remove all resources managed by the deployment from the lattice. Otherwise, previously managed resources will remain running. 
+
+**Note** The response is delivered immediately as an acknowledgement, as it can take several minutes to complete a "destructive" undeploy.
+
+**Request**: 
+```json
+{
+    "destructive": false
+}
+```
+
+**Response**:
+```json
+{
+    "result": "success|error",
+    "message": "..."
+}
+```
+
+### Deployment Status
+`wadm.api.{lattice}.model.status.{name}`
+
+Describes the deployment status of a given model. Note that since only one version of a model is ever deployed at one time, we don't need to specify a version here. The currently deployed version is included in the response.
```

Comment:
does status mean _planned_ status or actual status?  E.g., if model-v1 is runnning and you submit deploy model-v2 then immediaely request status, before v2 has been deployed, what would status tell us?

Reply:
actual status. there's no such thing as planned status. if you deploy v2 while v1 is running, the status will immediately switch to compensating (assuming there's a change required). This behavior is actually already described in the doc in the `deploy` operation.